test/common/http/http2/frame_replay.cc

```diff
@@ -67,29 +67,23 @@ CodecFrameInjector::CodecFrameInjector(const std::string& injector_name)
 }
 
 ClientCodecFrameInjector::ClientCodecFrameInjector() : CodecFrameInjector("server") {
-  auto client = std::make_unique<TestClientConnectionImpl>(client_connection_, client_callbacks_,
-                                                           stats_store_, settings_,
-                                                           Http::DEFAULT_MAX_REQUEST_HEADERS_KB);
-  request_encoder_ = &client->newStream(response_decoder_);
-  connection_ = std::move(client);
+  EXPECT_CALL(client_callbacks_, onGoAway()).Times(testing::AnyNumber());
```

Comment:
Done. I think it does make sense to set up a FakeClient/ServerConnection and a FakeStream. 

Not sure how much I should reuse from the integration test framework, since it seems like I don't need as much functionality as `FakeStream` and `FakeHttpConnection` provide. I think I want these fuzzers to only fuzz `codec_impl.cc` with minimal complexity outside of it, I would need:
- to implement `newStream` for connection callbacks (and probably use a new Fake stream per test iteration)
- noops / logging for MockConnection to handle watermark-related callbacks and `write(_,_)`

Reply:
I don't think it's necessary to use the fakes from the integration framework. I think the fakes we want here look a lot like the mocks; a nop implementation of the various interfaces (or with very simple logic to appear to do what we expect). The key difference is they wouldn't use the high cost mock framework (from performance perspective).

Out of interest, did you take a look to see if there is anything on the interwebs about performance of gmock?

Reply:
Yeah, I agree. There's very few methods from these objects that `codec_impl.cc` uses. The internet mostly talks about gMock performance with regard to a) compile time and b) removing the performance overhead of virtual methods (which would be to make a non-related class that doesn't inherit from the real object, and template the produce code.......)

I am really interested to know in the performance difference between initializing a mock object and the real object -- I'll see if I can get any more information.